File: Labs/Lab5(2.2)/python/big_letters.py
from abc_big_letters import ABC_BigLetters
from line import Line

class BigLetters(Line, ABC_BigLetters):
    def __init__(self, symbols):
        if type(symbols) == BigLetters:
            self.__line = BigLetters(symbols.get_symbols())
        elif type(symbols) == str: 
            self.__line:str = symbols.upper()
        else:
            self.__line = ''
    
    def get_len(self):
        return super().get_len(self.__line)
    
    def get_symbol_amount(self) -> str:
        return super().get_symbol_amount(self.__line, 'B')
    
    def get_symbols(self) -> str:
        return self.__line.upper()
    
    def __str__(self):
        return f"BigLetters: {self.__line}, Length: {self.get_len()}, Symbol 'B' Amount: {self.get_symbol_amount()}"

File: Labs/Lab5(2.2)/c#/Big_letters.cs
namespace LineApp
{
    public class BigLetters : Line
    {
        private readonly string _line;

        public BigLetters()
        {
            _line = string.Empty;
        }
        public BigLetters(string input)
        {
            _line = input;
        }

        public BigLetters(BigLetters other) : base()
        {
            _line = other.GetSymbols();
        }
        public override int GetLen(string line)
        {
            int uppercaseCount = 0;

            foreach (char c in line)
            {
                if (char.IsUpper(c))
                {
                    uppercaseCount++;
                }
            }
            return uppercaseCount;
        }

        public override int GetSymbolAmount(string line, char symbol = 'B')
        {
            return base.GetSymbolAmount(_line, symbol);
        }

        public string GetSymbols()
        {
            return _line;
        }

        public override string ToString()
        {
            return $"BigLetters: {_line}, Length: {GetLen(_line)}, Symbol 'B' Amount: {GetSymbolAmount(_line)}";
        }
    }
}